### PR TITLE
Add string generation

### PR DESCRIPTION
**Pull request by Hellwest**

### CODE REVIEW

File: os.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <stdlib.h>
```
Comment (Orlha) [inline os.h]: C++ project
```suggestion
#include <cstdlib>
```

File: os.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+#include <time.h>
+
+namespace os {
+	std::string random_sequence(int length) {
+		int i = 0;
+		int randomNumber;
+		char letter;
+		std::string result = "";
```
Comment (Orlha) [inline os.h]: std::string class is empty-string initialized by default;
```suggestion
		std::string result;
```

File: os.h [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <stdlib.h>
+#include <time.h>
+
+namespace os {
+	std::string random_sequence(int length) {
+		int i = 0;
```
Comment (Orlha) [inline os.h]: Local variable definition could be moved to for-loop head;
```suggestion

```

File: os.h [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+		int randomNumber;
```
Comment (Orlha) [inline os.h]: Local variable definition could be moved to for-loop head;
```suggestion
```

File: os.h [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+		char letter;
+		std::string result = "";
+
+		std::srand(time(NULL));
+		for (i = 0; i < length; i++) {
```
Comment (Orlha) [inline os.h]:
```suggestion
		for (int i = 0; i < length; ++i) {
```

File: os.h [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+			randomNumber = std::rand() % 26;
```
Comment (Orlha) [inline os.h]:
```suggestion
			int randomNumber = std::rand() % 26;
```

File: os.h [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+			letter = 'a' + randomNumber;
```
Comment (Orlha) [inline os.h]:
```suggestion
			char letter = 'a' + randomNumber;
```

File: os.h [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+			result[i] = letter;
```
Comment (Orlha) [inline os.h]:
```suggestion
			result += letter;
```

File: main.cpp [hunk 9]
```diff
@@ -1,3 +1,16 @@
...
+
+int main(int argc, char *argv[]) {
+	if (argc < 2) {
+		std::cout << "Provide a password length\n";
+		return -1;
+	}
+
+	int length = std::stoi(argv[1]);
+	std::string result = os::random_sequence(length);
+
```
Comment (Orlha) [inline main.cpp]:
```suggestion
std::cout << result << "\n";
```

**Review by Orlha — CHANGES_REQUESTED** [review]

File: os.h [hunk 10]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <stdlib.h>
+#include <time.h>
+
+namespace os {
```
Comment (Hellwest) [inline os.h]: `namespace oc {`

File: main.cpp [hunk 11]
```diff
@@ -1,3 +1,16 @@
-int main() {
+#include <iostream>
+#include <string>
+#include "os.h"
```
Comment (Hellwest) [inline main.cpp]: `#include <oc.h>`

File: .gitignore [hunk 12]
```diff
@@ -1 +1,3 @@
 build
+*.out
```
Comment (Orlha) [inline .gitignore:2]: When building via CMake, build generation and compilation processes are done in "build" folder, so further git-ignores are not necessary.

File: CMakeLists.txt [hunk 13]
```diff
@@ -8,6 +8,7 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
 set (sources
 	main.cpp
+	oc.h
```
Comment (Orlha) [inline CMakeLists.txt]: Headers are included, implementation is compiled.
```suggestion
	oc.cpp
```

File: oc.cpp [hunk 14]
```diff
@@ -0,0 +1,18 @@
+#include "oc.h"
+
+constexpr char OC::possible_chars[];
```
Comment (Orlha) [inline oc.cpp]: Not sure about the design choice here.
`constexpr static` in header should be enough.

File: oc.h [hunk 15]
```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <chrono>
```
Comment (Orlha) [inline oc.h]: If an interface doesn't depend on these parts of library, they can be safely moved to an implementation file.

**Pull request merged**